templates/hostname-RedHat.erb

```diff
@@ -10,4 +10,6 @@ NOZEROCONF="<%= @nozeroconf %>"
 NETWORKING_IPV6="<%= @ipv6enable %>"
 IPV6INIT="<%= @ipv6enable %>"
 <% end -%>
-HOSTNAME="<%= @manage_hostname.split('.').first %>"
+<% if @hostname -%>
+HOSTNAME="<%= @hostname %>"
+<% end -%>
```

Comment:
This would also fix #35 
I always set the full fqdn and not the short hostname only.